From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: saturn-linux contributors
Date: Mon, 31 Aug 2026 14:00:00 -0300
Subject: [PATCH] sh: add initial Sega Saturn SH-2 platform selection

This patch is intended to be applied on top of Linux v4.0.

The Saturn uses the existing generic SH-2 CPU support. This adds a
CPU subtype, Kconfig machine option, and machine-directory selection.
Hardware initialization and drivers are deliberately left for later
bring-up stages.

---
 arch/sh/Kconfig        | 8 ++++++++
 arch/sh/Makefile       | 1 +
 arch/sh/boards/Kconfig | 9 +++++++++
 3 files changed, 18 insertions(+)

diff --git a/arch/sh/Kconfig b/arch/sh/Kconfig
--- a/arch/sh/Kconfig
+++ b/arch/sh/Kconfig
@@ -90,6 +90,14 @@ choice
 # SH-2 Processor Support
 #
 
+config CPU_SUBTYPE_SEGA_SATURN
+	bool "Support Sega Saturn processor"
+	select CPU_SH2
+	help
+	  Select this option when building Linux for the Hitachi SH-2
+	  processor used by the Sega Saturn.
+
 config CPU_SUBTYPE_SH7619
 	bool "Support SH7619 processor"
 	select CPU_SH2

diff --git a/arch/sh/boards/Kconfig b/arch/sh/boards/Kconfig
--- a/arch/sh/boards/Kconfig
+++ b/arch/sh/boards/Kconfig
@@ -1,5 +1,13 @@
 menu "Board support"
 
+config SH_SATURN
+	bool "Sega Saturn"
+	depends on CPU_SUBTYPE_SEGA_SATURN
+	help
+	  Enable support for the Sega Saturn console platform.
+
 config SOLUTION_ENGINE
 	bool
@@ -180,6 +188,7 @@ config SH_APSH4AD0A
 source "arch/sh/boards/mach-sdk7780/Kconfig"
 source "arch/sh/boards/mach-migor/Kconfig"
 source "arch/sh/boards/mach-rsk/Kconfig"
+source "arch/sh/boards/mach-saturn/Kconfig"
 
 if SH_MAGIC_PANEL_R2

diff --git a/arch/sh/Makefile b/arch/sh/Makefile
--- a/arch/sh/Makefile
+++ b/arch/sh/Makefile
@@ -130,6 +130,7 @@ machdir-$(CONFIG_SH_CAYMAN)			+= mach-cayman
 machdir-$(CONFIG_SH_RSK)			+= mach-rsk
+machdir-$(CONFIG_SH_SATURN)			+= mach-saturn
 
 ifneq ($(machdir-y),)
 core-y	+= $(addprefix arch/sh/boards/, \
-- 
2.0.0
